### PR TITLE
Creating base classes Position, Interfaces and Pieces.

### PR DESCRIPTION
**Pull request by MirkoFer-622**
Creating base class for Position, ICheck, IPawn, IMovable, ICastling, Piece and other types of pieces; adding Main.java to see correct objects creation.

### CODE REVIEW

File: src/main/java/chess/Piece.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public void setPosition(final Position position) {
+        this.position = position;
+    }
+
+    public String toString() {
+        String initial;
+        if (this.color == Color.WHITE) {
+            initial = "W";
+        } else {
+            initial = "B";
```
Comment (roht0515) [inline src/main/java/chess/Piece.java]: why don't you use constants for "W" and "B"?

File: src/main/java/chess/Piece.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        }
+        return "" + this.symbol + initial;
+    }
```
Comment (roht0515) [inline src/main/java/chess/Piece.java]: it's not a good idea to convert the result to string adding double quotation marks -> ""

File: src/main/java/chess/Piece.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
+enum Color {
```
Comment (JhordanSoto) [inline src/main/java/chess/Piece.java]: Maybe,You can put the enum apart

File: src/main/java/chess/Piece.java (hunk 1 above)
Comment (MirkoFer-622) [reply src/main/java/chess/Piece.java]: I updated it right now, could you review it please?

File: src/main/java/chess/Piece.java (hunk 2 above)
Comment (MirkoFer-622) [reply src/main/java/chess/Piece.java]: I updated it right now, could you review it please?

File: src/main/java/chess/Piece.java (hunk 3 above)
Comment (MirkoFer-622) [reply src/main/java/chess/Piece.java]: I updated it right now, could you review it please?

**Pull request merged**